tests/test_api_import_guardrails.py

```
@@ -0,0 +1,272 @@
+3. Predictable behavior - no hidden initialization on import
+
+The API was refactored to avoid import side effects. These tests prevent
+regressions where module-level app creation is accidentally reintroduced.
+
+Related commit: 07e61d4 feat: Refactor API structure to avoid import side effects
+"""
+
+import importlib
+import sys
+from pathlib import Path
+from typing import List
```

Ruff (0.14.13)

`typing.List` is deprecated, use `list` instead

(UP035)